### PR TITLE
Enforce no-negated-condition lint rule, fix + test handful of offenders.

### LINKED ISSUE #698: Mirror SonarQube's negated-condition gate locally via ESLint no-negated-condition
SonarCloud's "Unexpected negated condition" rule caught a negated ternary in PR #694 (`status !== undefined ? … : ""`) that had already been pushed — the round-trip through CI is exactly the latency the local `sonarjs/cognitive-complexity` mirror was introduced to eliminate. We should mirror this SonarQube rule locally too, so `pnpm run lint` and the pre-push hook flag a negated condition before it reaches CI.

## Strategy

The SonarQube "Unexpected negated condition" check delegates to ESLint's **core** `no-negated-condition` rule — no new plugin or dependency required. Add it alongside the existing cognitive-complexity mirror in `eslint.config.mjs`:

```js
{
  files: ["src/**/*.ts"],
  ignores: ["**/*.test.ts", "**/*.d.ts"],
  rules: {
    "sonarjs/cognitive-complexity": ["error", 15],
    "no-negated-condition": "error",
  },
}
```

Scope it to non-test `src` to match `sonar.exclusions` (tests and generated `.d.ts` are excluded from SonarQube analysis), consistent with the cognitive-complexity block. `no-negated-condition` flags a negated `if`/ternary condition that has an `else` branch (e.g. `!a ? x : y`, `a !== b ? x : y`, `if (!a) {…} else {…}`) — the fix is always to flip to the positive-first form.

## Clean up existing violations

Enabling the rule may surface pre-existing negated conditions across `src`. As part of this issue, **flip each one to positive-first** (not suppress) and confirm a clean `pnpm run lint`. Verify the exact set SonarCloud already reports matches what the local rule flags; if the two diverge on an edge case, note it so the mirror stays faithful.

## Acceptance

- [ ] `no-negated-condition: "error"` added to the non-test `src` block in `eslint.config.mjs`
- [ ] Every violation the rule surfaces is fixed by flipping to positive-first (none suppressed)
- [ ] `pnpm run lint` passes clean; a deliberately-added negated ternary fails it (verify, then revert)

## Relationships

Motivated by a SonarCloud finding on PR #694. Same "mirror the SonarQube gate locally" motivation as the `sonarjs/cognitive-complexity` rule (epic #654), but a distinct rule — intentionally parentless.

### PR DESCRIPTION
**Pull request by jlrobins**
Enable the `no-negated-condition` lint rule, preventing it from becoming a SonarCube complaint on future PRs. Fix and ensure test coverage over the handful of existing offenders.

Closes #698

### CODE REVIEW

File: src/confluent/tools/handlers/flink/diagnostics/query-profiler-handler.test.ts [hunk 1]
```diff
@@ -223,6 +223,34 @@ describe("query-profiler-handler.ts", () => {
...
+        const cm = getMockedClientManager();
+        cm.getConfluentCloudFlinkRestClient().GET.mockResolvedValue({
+          data: { Graph: JSON.stringify(TASK_GRAPH) },
+        });
+        // task/busy_time_ms_per_second at 456 ms/s → 45.6%;
+        // task/idle_time_ms_per_second at 123 ms/s → 12.3%
+        wireFlinkTelemetry(cm, {
+          "task/busy_time": [{ value: 456, taskId: "task-1" }],
+          "task/idle_time": [{ value: 123, taskId: "task-1" }],
+        });
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/diagnostics/query-profiler-handler.test.ts:240]: Fixed.

**Review by bcscc — APPROVED** [review]
